### PR TITLE
Improve liniting for unused vars

### PR DESCRIPTION
**Pull request by clemens-msupply**
I had some false positive with the current configuration, hope this is better now...

### CODE REVIEW

File: client/.eslintrc.js [hunk 1]
```diff
@@ -33,8 +33,15 @@ module.exports = {
...
     'valid-jsdoc': 'off',
     'react/display-name': 'off',
-    'no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
-    '@typescript-eslint/no-unused-vars': 'off',
+    'no-unused-vars': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      {
+        argsIgnorePattern: '^_',
+        varsIgnorePattern: '^_',
```
Comment (CarlosNZ) [inline client/.eslintrc.js:41]: Just wondering why we'd want to ignore unused vars?

Comment (andreievg) [reply client/.eslintrc.js:41]: if we want like the second var but no the first

**Comment by andreievg** [comment]
hmm, the develop branch version works for me, and we have some `_parameters`, like: https://github.com/openmsupply/open-msupply/blob/8baa708b907db71463665bb1ec349729a2cbc51f/client/packages/common/src/authentication/AuthContext.tsx#L86-L88

File: client/.eslintrc.js (hunk 1 above)
Comment (CarlosNZ) [reply client/.eslintrc.js:41]: That's unused args, which we already have. Was wondering about the `vars`, specifically.

Comment (andreievg) [reply client/.eslintrc.js:41]: ohh true `varsIgnorePattern` 🏃‍♂️

Comment (clemens-msupply) [reply client/.eslintrc.js:41]: yes right no real reason to have it in

**Comment by clemens-msupply** [comment]
> hmm, the develop branch version works for me, and we have some `_parameters`, like:
> 
> https://github.com/openmsupply/open-msupply/blob/8baa708b907db71463665bb1ec349729a2cbc51f/client/packages/common/src/authentication/AuthContext.tsx#L86-L88

Thanks, will have a look with the dev version. @CarlosNZ and I had some cases before where it was not working as expected and made changes in the programs branch...

**Review by CarlosNZ — APPROVED** [review]
LGTM!

**Pull request merged**